src/interpreter.cc

```diff
@@ -116,6 +116,9 @@ namespace rego
 
   Node Interpreter::get_errors(const Node& node) const
   {
+    if (!node->get_and_reset_contains_error())
```

Comment:
You should check if node is null. I believe this may be the source of the segfaults.

Reply:
Fixed. It was a recursive call that was the problem.